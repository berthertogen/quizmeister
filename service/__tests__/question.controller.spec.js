const request = require('superagent');

const buildUrl = (path) => {
  const url =  `${process.env.SERVICE_URL}/question/${path}`;
  return `${process.env.SERVICE_URL}/question/${path}`;
};

test("should create question on Post", async () => {
  var response = await request
    .post(buildUrl(""))
    .send({
      title: 'title',
      remark: 'remark',
      type: 2,
      answers: [
        { text: 'answer 1', correct: true },
        { text: 'answer 2', correct: false }
      ],
      scoring: {
        type: 1,
        weightCorrectAnswer: 1,
        weightNoAnswer: 0,
        timeLimitSeconds: undefined,
        timeScoringInterval: undefined
      }
    })
    .set("Accept", "application/json");
  expect(response.headers["content-type"]).toBe("application/json; charset=utf-8")
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot({
    modifiedOn: expect.any(String),
    searchField: expect.any(String),
    shortId: expect.any(String),
  });
});

test("should get question on Get", async () => {
  var response = await request
    .get(buildUrl(`1`))
    .set("Accept", "application/json");
  expect(response.headers["content-type"]).toBe("application/json; charset=utf-8")
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot({
    modifiedOn: expect.any(String),
    searchField: expect.any(String),
    shortId: expect.any(String),
  });
});

test("should get questions on Get (search)", async () => {
  var response = await request
    .get(buildUrl(`search/0/20`))
    .set("Accept", "application/json");
  expect(response.headers["content-type"]).toBe("application/json; charset=utf-8")
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot([{
    modifiedOn: expect.any(String),
    searchField: expect.any(String),
    shortId: expect.any(String),
  }]);
});

test("should update question on Put", async () => {
  var response = await request
    .put(buildUrl(`1`))
    .send({
      title: 'updated title',
      remark: 'remark',
      type: 2,
      answers: [
        { answerId: 1, text: 'answer 1', correct: true },
        { answerId: 2, text: 'answer 2', correct: false }
      ],
      scoring: {
        scoringId: 1,
        type: 1,
        weightCorrectAnswer: 1,
        weightNoAnswer: 0,
        timeLimitSeconds: undefined,
        timeScoringInterval: undefined
      }
    })
    .set("Accept", "application/json");
  expect(response.headers["content-type"]).toBe("application/json; charset=utf-8")
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot({
    modifiedOn: expect.any(String),
    searchField: expect.any(String),
    shortId: expect.any(String),
  });
});

test("should delete question on Delete", async () => {
  var responseDelete = await request
    .delete(buildUrl(`1`))
    .set("Accept", "application/json");
  expect(responseDelete.headers["content-type"]).toBe("application/json; charset=utf-8")
  expect(responseDelete.status).toBe(200);
  var responseGet = await request
    .get(buildUrl(`search/0/20`))
  expect(responseGet.status).toBe(200);
  expect(responseGet.body).toMatchSnapshot();
});
