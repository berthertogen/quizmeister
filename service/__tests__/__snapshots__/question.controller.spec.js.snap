// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create question on Post 1`] = `
Object {
  "answers": Array [
    Object {
      "answerId": 1,
      "correct": true,
      "searchField": "trueanswer 1",
      "text": "answer 1",
    },
    Object {
      "answerId": 2,
      "correct": false,
      "searchField": "falseanswer 2",
      "text": "answer 2",
    },
  ],
  "modifiedOn": Any<String>,
  "questionId": 1,
  "remark": "remark",
  "rounds": null,
  "scoring": Object {
    "scoringId": 1,
    "timeLimitSeconds": 0,
    "timeScoringInterval": 0,
    "type": 1,
    "weightCorrectAnswer": 1,
    "weightNoAnswer": 0,
  },
  "searchField": Any<String>,
  "shortId": Any<String>,
  "title": "title",
  "type": 2,
}
`;

exports[`should delete question on Delete 1`] = `Array []`;

exports[`should get question on Get 1`] = `
Object {
  "answers": Array [
    Object {
      "answerId": 1,
      "correct": true,
      "searchField": "trueanswer 1",
      "text": "answer 1",
    },
    Object {
      "answerId": 2,
      "correct": false,
      "searchField": "falseanswer 2",
      "text": "answer 2",
    },
  ],
  "modifiedOn": Any<String>,
  "questionId": 1,
  "remark": "remark",
  "rounds": Array [],
  "scoring": Object {
    "scoringId": 1,
    "timeLimitSeconds": 0,
    "timeScoringInterval": 0,
    "type": 1,
    "weightCorrectAnswer": 1,
    "weightNoAnswer": 0,
  },
  "searchField": Any<String>,
  "shortId": Any<String>,
  "title": "title",
  "type": 2,
}
`;

exports[`should get questions on Get (search) 1`] = `
Array [
  Object {
    "answers": Array [
      Object {
        "answerId": 1,
        "correct": true,
        "searchField": "trueanswer 1",
        "text": "answer 1",
      },
      Object {
        "answerId": 2,
        "correct": false,
        "searchField": "falseanswer 2",
        "text": "answer 2",
      },
    ],
    "modifiedOn": Any<String>,
    "questionId": 1,
    "remark": "remark",
    "rounds": null,
    "scoring": Object {
      "scoringId": 1,
      "timeLimitSeconds": 0,
      "timeScoringInterval": 0,
      "type": 1,
      "weightCorrectAnswer": 1,
      "weightNoAnswer": 0,
    },
    "searchField": Any<String>,
    "shortId": Any<String>,
    "title": "title",
    "type": 2,
  },
]
`;

exports[`should update question on Put 1`] = `
Object {
  "answers": Array [
    Object {
      "answerId": 1,
      "correct": true,
      "searchField": "trueanswer 1",
      "text": "answer 1",
    },
    Object {
      "answerId": 2,
      "correct": false,
      "searchField": "falseanswer 2",
      "text": "answer 2",
    },
  ],
  "modifiedOn": Any<String>,
  "questionId": 1,
  "remark": "remark",
  "rounds": null,
  "scoring": Object {
    "scoringId": 1,
    "timeLimitSeconds": 0,
    "timeScoringInterval": 0,
    "type": 1,
    "weightCorrectAnswer": 1,
    "weightNoAnswer": 0,
  },
  "searchField": Any<String>,
  "shortId": Any<String>,
  "title": "updated title",
  "type": 2,
}
`;
